mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -75,35 +92,71 @@ sync_repo() {
         echo "SHAs haven't changed!"
         return
     fi
-    git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" commit -m "sync(k8s.io/kubernetes): ${newKubeSHA}" -- kubernetes-sha
+    git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" commit -m "sync(k8s.io/kubernetes) ${newKubeSHA}" -- kubernetes-sha
+}
+
+# deduplicate_commits reset branch to origin/branch if they are the same.
+# This function assumes the work directory is the repo that needs to deduplication.
+deduplicate_commits() {
```

Comment:
This function is added to prevent the robot from creating a pair of `sync: reset Godeps.json` and `sync: resync vendor folder` commits that in combine is a no-op.